File: android/app/src/main/java/net/dailyok/android/network/DailyOKError.kt
package net.dailyok.android.network

sealed class DailyOKError : Exception() {
    data class Network(override val message: String = "Network error. Please check your connection.") : DailyOKError()
    data class Auth(override val message: String = "Authentication failed. Please sign in again.") : DailyOKError()
    data class NotFound(override val message: String = "The requested resource was not found.") : DailyOKError()
    data class ServerError(override val message: String = "Server error. Please try again later.") : DailyOKError()
    data class Offline(override val message: String = "You are offline. Your action will sync when reconnected.") : DailyOKError()
    data class Unknown(override val message: String = "An unexpected error occurred.") : DailyOKError()

    val localizedMessage: String
        get() = message ?: "An unexpected error occurred."
}

File: ios/DailyOK/Models/DailyOKError.swift
import Foundation

/// Centralized error type for all Daily OK services.
/// Provides user-facing localized descriptions for consistent error display.
enum DailyOKError: LocalizedError {
    case network(Error)
    case auth(String)
    case notFound(String)
    case serverError(String)
    case offline
    case unknown(Error)

    var errorDescription: String? {
        switch self {
        case .network(let error):
            return String(localized: "Network error: \(error.localizedDescription)")
        case .auth(let message):
            return message
        case .notFound(let resource):
            return String(localized: "\(resource) not found.")
        case .serverError(let message):
            return String(localized: "Server error: \(message)")
        case .offline:
            return String(localized: "You appear to be offline. Your action will be completed when you reconnect.")
        case .unknown(let error):
            return error.localizedDescription
        }
    }
}
